diff --git a/node_modules/@metamask/transaction-controller/dist/TransactionController.cjs b/node_modules/@metamask/transaction-controller/dist/TransactionController.cjs
index 206d773..1eb8cf8 100644
--- a/node_modules/@metamask/transaction-controller/dist/TransactionController.cjs
+++ b/node_modules/@metamask/transaction-controller/dist/TransactionController.cjs
@@ -95,6 +95,12 @@ const metadata = {
         includeInDebugSnapshot: false,
         usedInUi: false,
     },
+    swapsTransactions: {
+        includeInStateLogs: true,
+        persist: true,
+        includeInDebugSnapshot: false,
+        usedInUi: false,
+    },
 };
 /**
  * Multiplier used to determine a transaction's increased gas fee during cancellation
